src/kvstore/RocksEngine.cpp

```diff
@@ -153,6 +154,7 @@ ResultCode RocksEngine::get(const std::string& key, std::string* value) {
 std::vector<Status> RocksEngine::multiGet(const std::vector<std::string>& keys,
                                           std::vector<std::string>* values) {
     rocksdb::ReadOptions options;
+    options.total_order_seek = true;
```

Comment:
total_order_seek with prefix bloom filter will result in huge performance issue.

Reply:
Code in BlockBasedTable::NewIterator
```
    return new BlockBasedTableIterator<DataBlockIter>(
        this, read_options, rep_->internal_comparator,
        NewIndexIterator(
            read_options,
            need_upper_bound_check &&
                rep_->index_type == BlockBasedTableOptions::kHashSearch,
            /*input_iter=*/nullptr, /*get_context=*/nullptr, &lookup_context),
        !skip_filters && !read_options.total_order_seek &&
            prefix_extractor != nullptr,
        need_upper_bound_check, prefix_extractor, BlockType::kData, caller,
        compaction_readahead_size);
```
According to code，if total_order_seek = true, means we don't consider prefix bloom filter, so there won't be any performance issue?

Reply:
IIRC, total_order_seek need to be true when we use prefix bloom filter? Of course, we can set it according to `FLAGS_enable_prefix_bloom_filter`

Reply:
If we implement custom Transformer,  we could set it to be false.

Reply:
The whole key filters for Get() and MultiGet is enough, so it's better to set it to be true to disable the prefix filter?

Reply:
Make sense.

Reply:
So can block-based table assure all data in order when prefix bloom filter enabled?

Reply:
Rocksdb can guaranteed a total order of keys for the same prefix.

Reply:
According the docs of rocksdb, total_order_seek will ensure the behavior the same as no prefix bloom filter, by ordering the index. Benchmark may be required for all prefix operation with total_order_seek on.